test/algebra/fields/plonk/logic_and_flag.cpp

```diff
@@ -0,0 +1,166 @@
+
+    typename BlueprintFieldType::value_type p = public_input[0]*public_input[1];
+    typename BlueprintFieldType::value_type expected_result = (p.is_zero() ? BlueprintFieldType::value_type::zero() : BlueprintFieldType::value_type::one());
+
+    auto result_check = [&expected_result, &public_input](AssignmentType &assignment,
+	    typename component_type::result_type &real_res) {
+            #ifdef BLUEPRINT_PLONK_PROFILING_ENABLED
+            std::cout << "logic and test: \n";
+            std::cout << "input   : " << public_input[0].data << " " << public_input[1].data << "\n"; 
+            std::cout << "expected: " << expected_result.data    << "\n";
+            std::cout << "real    : " << var_value(assignment, real_res.output).data << "\n\n"; 
+            #endif
+            assert(var_value(assignment, real_res.output) == expected_result);
+    };
+
+
+    nil::crypto3::test_component<component_type, BlueprintFieldType, ArithmetizationParams, hash_type, Lambda>(
+        component_instance, public_input, result_check, instance_input);
+}
+
+template<typename FieldType, std::size_t RandomTestsAmount, std::uint32_t WitnessesAmount>
+void test_logic_and_flag_random_input_and_zero(){
+
+    nil::crypto3::random::algebraic_engine<FieldType> generate_random;
+    boost::random::mt19937 seed_seq;
+    generate_random.seed(seed_seq);
+
+    typename FieldType::value_type input = generate_random();
+    typename FieldType::integral_type input_integral = typename FieldType::integral_type(input.data);
+    input_integral = input_integral & 0xFFFFFFFFFFFFFFFF_cppui255;
+    typename FieldType::value_type input_scalar =  input_integral;
+    test_logic_and_flag<FieldType, WitnessesAmount>({input_scalar, 0});
+    test_logic_and_flag<FieldType, WitnessesAmount>({0, input_scalar});
+}
+
+
+
+template<typename FieldType, std::size_t RandomTestsAmount, std::uint32_t WitnessesAmount>
+void test_range_check_random_inputs(){
+
+    nil::crypto3::random::algebraic_engine<FieldType> generate_random;
+    boost::random::mt19937 seed_seq;
+    generate_random.seed(seed_seq);
+
+    for (std::size_t i = 0; i < RandomTestsAmount; i++){
+        typename FieldType::value_type input_x = generate_random();
+    	typename FieldType::integral_type input_integral_x = typename FieldType::integral_type(input_x.data);
+        input_integral_x = input_integral_x & 0xFFFFFFFFFFFFFFFF_cppui255;
+
+        typename FieldType::value_type input_y = generate_random();
+    	typename FieldType::integral_type input_integral_y = typename FieldType::integral_type(input_y.data);
+        input_integral_y = input_integral_y & 0xFFFFFFFFFFFFFFFF_cppui255;
+
+        test_logic_and_flag<FieldType, WitnessesAmount>({input_integral_x,input_integral_y}); 
+	}
+}
+
+
+constexpr static const std::size_t random_tests_amount = 1;
+
+BOOST_AUTO_TEST_SUITE(blueprint_plonk_logic_and_test_suite)
+
+#define test_case_expand(WitnessesAmount) \
```

Comment:
test_case_expand can be a function, and not a macro (in both test files). Probably would also have to be renamed?